HW05.Rmd

```diff
@@ -0,0 +1,133 @@
+---
+title: "HW05"
+output: github_document
+author: Bei Wang
+---
+
+```{r load pkgs, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library("tidyverse")
+library("magrittr")
+library("knitr")
+library("readr")
+```
+
+## 1. Tidying the dad_mom file
+
+    Each variable must have its own column.
+    Each observation must have its own row.
+    Each value must have its own cell.
+
+```{r dad_mom}
+#read dataset
+dad_mom<- read.table("dad_mom.txt", header=TRUE) 
+
+#reshape
+tidy_dad_mom<- dad_mom %>%
+  unite("dad", name_dad, income_dad) %>%
+  unite("mom", name_mom, income_mom) %>%
+  pivot_longer(c(dad, mom), names_to="parent", values_to="income") %>%
+  separate(income, c("name", "income"))
+
+#print
+tidy_dad_mom
+```
+
+## 2. Joining together CTRP data
+
+```{r totalauc}
+#read files
+AUC_1 <- read_csv("CTRP_files/AUC_1.csv")
+AUC_2 <- read_csv("CTRP_files/AUC_2.csv")
+cancer_cell_line_info <- read_csv("CTRP_files/cancer_cell_line_info.csv")
+compound_info <- read_csv("CTRP_files/compound_info.csv")
+Experiment_info <- read_csv("CTRP_files/Experiment_info.csv")
+
+#glimpse
+glimpse(AUC_1)
+glimpse(AUC_2)
+glimpse(cancer_cell_line_info)
+glimpse(compound_info)
+glimpse(Experiment_info)
+
+#Since AUC_1 and AUC_2 look like two parrelle data frame with links to the other files, so i join AUC_1 and _2 first, and then combin them with the other files
+
+#join AUC_1 with _2
+AUC<- AUC_1 %>%
+  bind_rows(AUC_2)
+
+#join auc with other files
+total_AUC<- AUC %>%
+  inner_join(Experiment_info, by=c("experiment_id"="expt_id")) %>%
+  inner_join(compound_info, by="master_cpd_id") %>%
+  inner_join(cancer_cell_line_info, by="master_ccl_id") 
```

Comment:
When I used inner_join to join the dataset, I noticed that rows which don't have common master_cpd_id, master_ccl_id or experiment ids will be dropped and not included in the final dataset. I tried full_join() to solve this problem, and I guess maybe this is why you didn't get the results you need in the second and third questions...?

Reply:
That's a good call, although I experiment with either using "full_join" or, as Mingyu suggested, to use "str_detect" instead of "==" in my filter(), it turns out I should use "str_detect".